assests/javascript/index.js

```diff
@@ -136,6 +151,125 @@ async function updateTaskStatus(id, isDone) {
   }
 }
 
+function handleTitle() {
+  const editTaskIcons = Array.from(document.getElementsByClassName("edit-task-icon"));
+
+  editTaskIcons.forEach((editTaskIcon) => {
+    editTaskIcon.addEventListener("click", (event) => {
+      const id = event.target.parentNode.id;
+      const titleBox = document.getElementById(`title-box-${id}`);
+      const taskTitle = titleBox.innerText;
+
+      convertEditIconToSubmitIcon(id);
+      convertTaskTitleToInput(id, titleBox, taskTitle);
+      editTaskTitle(id, taskTitle);
+    })
+  })
+}
+
+function convertEditIconToSubmitIcon(id) {
+  hideEditIcon(id);
+  showSubmitIcon(id);
+}
+
+function hideEditIcon(id) {
+  const editIcon = document.getElementById(`edit-task-icon-${id}`);
+
+  editIcon.classList.remove("edit-task-icon");
+  editIcon.classList.add("hidden");
+}
+
+function showSubmitIcon(id) {
+  const submitIcon = document.getElementById(`submit-task-icon-${id}`);
+
+  submitIcon.classList.remove("hidden");
+  submitIcon.classList.add("submit-task-icon");
+}
+
+function convertTaskTitleToInput(id, titleBox, taskTitle) {
+  titleBox.innerHTML = `<input class="edit-task-title" id="edit-title-${id}" type="text" value="${taskTitle}"/>
+                        <p id="edit-error-${id}" class="edit-title-error"></p>`;
+}
+
+function editTaskTitle(id, taskTitle) {
+  const submitIcon = document.getElementById(`submit-task-icon-${id}`);
+
+  submitIcon.addEventListener("click", function (event) {
+    const newTaskTitle = document.querySelector('input[type=text]').value;
```

Comment:
The selector is very generic. Will select all inputs in the page.

Reply:
I will replace it by `className`